stellar/gateway.js

```diff
@@ -0,0 +1,210 @@
+                    weight: 1,
+                },
+            ],
+            threshold: 1,
+        };
+    }
+
+    return await getAmplifierVerifiers(config, chain.axelarId);
+};
+
+function encodeDataHash(commandType, command) {
+    const data = nativeToScVal([commandTypeToScVal(commandType), command]);
+    return keccak256('0x' + data.toXDR('hex'));
+}
+
+function getProof(dataHash, wallet, chain, options) {
+    const nonce = options.currentNonce ? arrayify(id(options.currentNonce)) : Array(32).fill(0);
+    const signers = weightedSignersToScVal({
+        nonce,
+        signers: [
+            {
+                signer: wallet.publicKey(),
+                weight: 1,
+            },
+        ],
+        threshold: 1,
+    });
+    const signerHash = keccak256(signers.toXDR());
+
+    const domainSeparator = chain.contracts.axelar_auth_verifier?.initializeArgs?.domainSeparator;
+
+    if (!domainSeparator) {
+        throw new Error('Domain separator not found');
+    }
+
+    const msg = '0x' + domainSeparator + signerHash.slice(2) + dataHash.slice(2);
+    const messageHash = keccak256(msg);
+    const signature = wallet.sign(arrayify(messageHash));
+
+    return proofToScVal({
+        signers: [
+            {
+                signer: {
+                    signer: wallet.publicKey(),
+                    weight: 1,
+                },
+                signature,
+            },
+        ],
+        threshold: 1,
+        nonce,
+    });
+}
+
+async function callContract(wallet, _, chain, contractConfig, args, options) {
+    const contract = new Contract(contractConfig.address);
+    const caller = nativeToScVal(Address.fromString(wallet.publicKey()), { type: 'address' });
+
+    const [destinationChain, destinationAddress, payload] = args;
+
+    const operation = contract.call(
+        'call_contract',
+        caller,
+        nativeToScVal(destinationChain, { type: 'string' }),
+        nativeToScVal(destinationAddress, { type: 'string' }),
+        nativeToScVal(Buffer.from(arrayify(payload)), { type: 'bytes' }),
+    );
+
+    await broadcast(operation, wallet, chain, 'Contract called', options);
+}
+
+async function approve(wallet, _, chain, contractConfig, args, options) {
+    const contract = new Contract(contractConfig.address);
+    const [sourceChain, messageId, sourceAddress, destinationAddress, payload] = args;
+
+    const messages = messagesToScVal([
+        {
+            messageId,
+            sourceChain,
+            sourceAddress,
+            contractAddress: destinationAddress === 'wallet' ? wallet.publicKey() : destinationAddress,
+            payloadHash: arrayify(keccak256(payload)),
+        },
+    ]);
+
+    const dataHash = encodeDataHash('ApproveMessages', messages);
+    const proof = getProof(dataHash, wallet, chain, options);
+
+    const operation = contract.call('approve_messages', messages, proof);
+
+    await broadcast(operation, wallet, chain, 'Approved Messages', options);
```

Comment:
Compared to `call_contract` & `Contract called`, this might be better to change it `approve_messages` & `Messages approved` or `Messages Approved`?